src/cloudant/replicator.py

```
@@ -0,0 +1,171 @@
+            target database, possibly including authentication info.
+        @param dict user_ctx: User to act as.
+
+        Additional params you might specify (I won't pass these along
+            unless specified):
+        @param boolean: create_target: specifies whether or not to
+            create the target, if it doesn't already exist.
+        @param boolean continuous: set to True for a continuous replication.
+
+        """
+
+        data = dict(
+            _id=repl_id if repl_id else unicode(uuid.uuid4()),
+            **kwargs
+        )
+
+        if not data.get('source'):
+            if source_db is None:
+                raise CloudantException(
+                    u"You must specify either a source_db Database "
+                    u"object or a manually composed 'source' string/dict."
+                )
+            data['source'] = {
+                "url": source_db.database_url,
+                "headers": {
+                    "Authorization": source_db.creds['basic_auth']
+                }
+            }
+
+        if not data.get('target'):
+            if target_db is None:
+                raise CloudantException(
+                    u"You must specify either a target_db Database "
+                    u"object or a manually composed 'target' string/dict."
+                )
+            data['target'] = {
+                "url": target_db.database_url,
+                "headers": {
+                    "Authorization": target_db.creds['basic_auth']
+                }
+            }
+
+        if not data.get('user_ctx'):
+            data['user_ctx'] = self.creds['user_ctx']
+
+        return self.create_document(data, throw_on_exists=True)
+
+    def list_replications(self):
+        """
+        _list_replications_
+
+        Returns a list of all replications.
+
+        """
+        docs = self.all_docs(include_docs="true")['rows']
+        return [doc['doc'] for doc in docs]
+
+    def replication_state(self, repl_id):
```

jcounts: in that case, we return None (and the test verifies that we do ... or, at least, considers that a valid return). When I think about couch docs, I generally think in JavaScript, so I think of a nonexistent field as simply having a null value, and "None" reflects that in Python.

We could raise an Exception, but that feels less correct to me ...